apps/topicprio/views.py

```diff
@@ -0,0 +1,147 @@
+import django_filters
+from django.contrib import messages
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+
+
+from adhocracy4.filters.filters import DefaultsFilterSet
+from adhocracy4.modules import views as module_views
+from adhocracy4.rules import mixins as rules_mixins
+
+from apps.contrib import filters
+from apps.dashboard.mixins import DashboardBaseMixin
+
+from . import forms
+from . import models
+
+
+class TopicFilterSet(DefaultsFilterSet):
+
+    defaults = {
+        'ordering': '-positive_rating_count'
+    }
+
+    category = filters.CategoryFilter()
+
+    ordering = filters.OrderingFilter(
+        choices=(
+            ('-positive_rating_count', _('Most popular')),
+            ('-comment_count', _('Most commented'))
+        )
+    )
+
+    class Meta:
+        model = models.Topic
+        fields = ['category']
+
+
+class TopicListView(module_views.ItemListView):
+    model = models.Topic
+    filter_set = TopicFilterSet
+
+    def get_queryset(self):
+        return super().get_queryset().filter(module=self.module) \
+            .annotate_positive_rating_count() \
+            .annotate_negative_rating_count() \
+            .annotate_comment_count()
+
+
+class TopicDetailView(module_views.ItemDetailView):
+    model = models.Topic
+    queryset = models.Topic.objects.annotate_positive_rating_count()\
+        .annotate_negative_rating_count()
+    permission_required = 'meinberlin_topicprio.view_topic'
+
+
+class TopicCreateFilterSet(django_filters.FilterSet):
+
+    category = filters.CategoryFilter()
+
+    class Meta:
+        model = models.Topic
+        fields = ['category']
+
+
+class TopicMgmtView(DashboardBaseMixin,
+                    rules_mixins.PermissionRequiredMixin,
+                    module_views.ItemListView):
```

Comment:
we could share code between managment views, as i had the same problem here: https://github.com/liqd/a4-meinberlin/blob/master/apps/polls/views.py#L46